navigation#3

VyrostkoOlga opened the pull request:
#3

fix: return to the screen with notes' list using pop (instead of push)
fix: update of the screen with notes' list

Commit:
navigation#3
fix: return to the screen with notes' list using pop (instead of push)
fix: update of the screen with notes' list

Commit:
navigation#3
fix: header of back button and navigation bar on the detailed view screen

mrdekk commented on NNotes/ViewController.h:
```diff
@@ -9,11 +9,13 @@
-@property (nonatomic, retain) NotesDataController * dataCtrl;
+@property (nonatomic, strong) NSIndexPath * index;
+@property (nonatomic, strong) NotesDataController * dataCtrl;
+@property (nonatomic, weak) NotesTableViewController * parent;
```
Так делать не стоит - лучше реализовать протокол во ViewController'е а NotesTableViewController в .m будет реализовывать этот протокол и устанавливать себя в качестве delegate'а

mrdekk commented on NNotes/ViewController.m:
```diff
@@ -111,30 +115,29 @@ - (IBAction)addOrEditNote:(id)sender {
-    ViewController * tableViewController = (ViewController *)[storyboard  instantiateViewControllerWithIdentifier:@"NotesTableViewController"];
-    
-    [self.navigationController pushViewController:tableViewController animated:YES];
+    [self.navigationController popToRootViewControllerAnimated: YES];
```
Здесь и далее лучше делать popViewControllerAnimated
потому что перед NotesTable... могут быть другие контроллеры у которых может быть несохраненное состояние и если мы "едем" на начальный контроллер мы это состояние можем потерять

Commit:
navigation#3
fix: usage of method popViewController instead of popToRootViewController

Commit:
navigation#3
fix: usage of Protocol and delegation instead of link to a parent view for a detailed view screen's Controller

mrdekk commented:
OK
